src/array/cuda/spmat_op_impl_csr.cu

```diff
@@ -228,17 +228,29 @@ template CSRMatrix CSRSliceRows<kDLGPU, int64_t>(CSRMatrix, int64_t, int64_t);
 template <typename IdType, typename DType>
 __global__ void _SegmentCopyKernel(
     const IdType* indptr, const DType* data,
-    const IdType* row, int64_t row_stride, int64_t length,
+    const IdType* row, int64_t length, int64_t n_row,
     const IdType* out_indptr, DType* out_data) {
-  int tx = blockIdx.x * blockDim.x + threadIdx.x;
+  IdType tx = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
Since `blockIdx`, `blockDim`, and `threadIdx` all contains `int`s, the right hand side here will evaluate to an `int`. You probably want to cast `blockIdx.x` to an `IdType`, so that all following operations will be promoted if `IdType` == `int64_t`:
```
IdType tx = static_cast<IdType>(blockIdx.x)*blockDIm.x + threadIdx.x;
```